instrumentation/kafka/kafka-clients/kafka-clients-common/library/src/main/java/io/opentelemetry/instrumentation/kafkaclients/OpenTelemetryKafkaMetrics.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.kafkaclients;
+
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.instrumentation.kafkaclients.internal.OpenTelemetryMetricsReporter;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.kafka.clients.CommonClientConfigs;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.common.metrics.MetricsReporter;
+
+/**
+ * Record kafka client metrics to OpenTelemetry.
+ *
+ * @see #getConfigProperties(OpenTelemetry)
+ */
+public class OpenTelemetryKafkaMetrics {
+
+  /**
+   * Produces a set of kafka client config properties (consumer or producer) to register a {@link
+   * MetricsReporter} that records metrics to the {@code openTelemetry} instance. Add these
+   * resulting properties to the configuration map used to initialize a {@link KafkaConsumer} or
+   * {@link KafkaProducer}.
+   *
+   * <p>For producers:
+   *
+   * <pre>{@code
+   * //    Map<String, Object> config = new HashMap<>();
+   * //    config.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, ...);
+   * //    config.putAll(OpenTelemetryKafka.openTelemetryConfigProperties(openTelemetry);
+   * //    try (KafkaProducer<?, ?> producer = new KafkaProducer<>(config)) { ... }
+   * }</pre>
+   *
+   * <p>For consumers:
+   *
+   * <pre>{@code
+   * //    Map<String, Object> config = new HashMap<>();
+   * //    config.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, ...);
+   * //    config.putAll(OpenTelemetryKafka.openTelemetryConfigProperties(openTelemetry);
+   * //    try (KafkaConsumer<?, ?> consumer = new KafkaConsumer<>(config)) { ... }
+   * }</pre>
+   *
+   * @param openTelemetry the {@link OpenTelemetry} used to record metrics
+   * @return the kafka client properties
+   */
+  public static Map<String, ?> getConfigProperties(OpenTelemetry openTelemetry) {
```

Comment:
Per the conversation in the 6/30 SIG meeting, I've adjusted the configuration pattern to not rely on `GlobalOpenTelemetry`. Instead, users use this to get a set of properties they should merge with their consumer / producer config.

`OpenTelemetryMetricsReporter#configure(Map<String, ?>)` receives the properties after construction, extracts the `OpenTelemetry` instance, and sets up the meter.

Reply:
We have `KafkaTelemetry` as an entrypoint already so we can probably merge this into it, it would stop being static / accepting an `OpenTelemetry`

Reply:
Kept it separate so that the instrumentation wasn't limited to kafka-clients version 2.6. Don't feel strongly about this though and happy to move it.

Reply:
Oh I didn't notice this was in `kafka-clients-common`, GitHub doesn't render long paths well :) That module isn't meant for direct consumption by users so this definitely needs to move. It would either move to a new kafka-clients-0.11/library or the current 2.6. 4/2020 release date for 2.6 isn't super-old, but I suppose it should still be ok.

Reply:
How about I keep the internal methods in `kafka-clients-common` and move the entry point to `KafkaTelemetry` in `kafka-client-2.6`? There currently isn't a library module for `kafka-client-0.11`. We can omit an entry point for `kafka-client-0.11` for now and add it when users ask for it.